packages/frontend/src/components/common/Filters/FilterInputs/DateFilterInputs.tsx

```diff
@@ -206,15 +195,8 @@ const DateFilterInputs = <T extends ConditionalRule = DateFilterRule>(
                         onChange={(v: Date | null) => {
                             onChange({
                                 ...rule,
-                                values:
-                                    v === null
-                                        ? []
-                                        : [
-                                              formatTimestamp(
-                                                  v,
-                                                  TimeFrames.SECOND,
-                                              ),
-                                          ],
+                                // format as an ISO string, not for display
+                                values: v === null ? [] : [dayjs(v).format()],
```

Comment:
We should *always* pass time around either as a Date object or the ISO string you get from format().